clustered/common/src/main/java/org/ehcache/clustered/common/messages/ServerStoreOpCodec.java

```diff
@@ -0,0 +1,130 @@
+        encodedMsg.putInt(cacheIdLen);
+        encodedMsg.put(message.getCacheId().getBytes(UTF_8));
+        encodedMsg.putLong(message.getKey());
+        encodedMsg.put(message.operation().getStoreOpCode());
+        encodedMsg.put(getAndAppendMessage.getPayload());
+        return encodedMsg.array();
+      case REPLACE:
+        ReplaceAtHeadMessage replaceAtHeadMessage = (ReplaceAtHeadMessage)message;
+        byte[] encodedExpectedChain = ChainCodec.encode(replaceAtHeadMessage.getExpect());
+        byte[] encodedUpdatedChain = ChainCodec.encode(replaceAtHeadMessage.getUpdate());
+        encodedMsg = ByteBuffer.allocate(MSG_TYPE_OFFSET + STORE_OP_CODE_OFFSET + CACHE_ID_LEN_OFFSET + KEY_OFFSET + cacheIdLen +
+              2 * CHAIN_LEN_OFFSET + encodedExpectedChain.length + encodedUpdatedChain.length);
+        encodedMsg.put(EhcacheEntityMessage.Type.SERVER_STORE_OP.getOpCode());
+        encodedMsg.putInt(cacheIdLen);
+        encodedMsg.put(message.getCacheId().getBytes(UTF_8));
+        encodedMsg.putLong(message.getKey());
+        encodedMsg.put(message.operation().getStoreOpCode());
+        encodedMsg.putInt(encodedExpectedChain.length);
+        encodedMsg.put(encodedExpectedChain);
+        encodedMsg.putInt(encodedUpdatedChain.length);
+        encodedMsg.put(encodedUpdatedChain);
+        return encodedMsg.array();
+      default:
+        throw new UnsupportedOperationException("This operation is not supported" + message.operation());
+    }
+  }
+
+  public static EhcacheEntityMessage decode(byte[] payload) {
+    ByteBuffer msg = ByteBuffer.wrap(payload);
+    msg.get();
+    int cacheIdLen = msg.getInt();
+    byte[] idArr = new byte[cacheIdLen];
+    msg.get(idArr);
+    String cacheId = new String(idArr, UTF_8);
+    long key = msg.getLong();
+    byte opCode = msg.get();
+    byte[] remaining = new byte[msg.remaining()];
+    msg.get(remaining);
+    switch (opCode) {
+      case 0:
+        return EhcacheEntityMessage.getOperation(cacheId, key);
+      case 1:
+        return EhcacheEntityMessage.getAndAppendOperation(cacheId, key, ByteBuffer.wrap(remaining).asReadOnlyBuffer());
+      case 2:
+        return EhcacheEntityMessage.appendOperation(cacheId, key, ByteBuffer.wrap(remaining).asReadOnlyBuffer());
+      case 3:
+        ByteBuffer replaceBuf = ByteBuffer.wrap(remaining);
+        int expectChainLen = replaceBuf.getInt();
+        byte[] encodedExpectChain = new byte[expectChainLen];
+        replaceBuf.get(encodedExpectChain);
+        int updateChainLen = replaceBuf.getInt();
+        byte[] encodedUpdateChain = new byte[updateChainLen];
+        replaceBuf.get(encodedUpdateChain);
+        return EhcacheEntityMessage.replaceAtHeadOperation(cacheId, key, ChainCodec.decode(encodedExpectChain),
+            ChainCodec.decode(encodedUpdateChain));
+      default:
+        throw new UnsupportedOperationException("This operation code is not supported" + opCode);
```

Comment:
Please pay attention to the message's formatting: you miss at least a space and ideally a space-colon-space (`' : '`) before concatenating the opcode.